src/components/productView/productViewContext.js

```diff
@@ -0,0 +1,312 @@
+
+/**
+ * Return the graph query based off of tally and capacity.
+ *
+ * @param {object} options
+ * @param {string} options.queryType
+ * @param {object} options.schemaCheck
+ * @param {object} options.options
+ * @returns {object}
+ */
+const useProductGraphTallyQuery = ({
+  queryType = 'graphTallyQuery',
+  schemaCheck = rhsmApiTypes.RHSM_API_QUERY_SET_REPORT_CAPACITY_TYPES,
+  options
+} = {}) =>
+  reduxHelpers.setApiQuery(
+    {
+      ...useProductQuery(),
+      ...useProductQueryFactory(queryType, options)
+    },
+    schemaCheck
+  );
+
+/**
+ * Return the inventory query for guests.
+ *
+ * @param {object} options
+ * @param {string} options.queryType
+ * @param {object} options.schemaCheck
+ * @param {object} options.options
+ * @returns {object}
+ */
+const useProductInventoryGuestsQuery = ({
+  queryType = 'inventoryGuestsQuery',
+  schemaCheck = rhsmApiTypes.RHSM_API_QUERY_SET_INVENTORY_GUESTS_TYPES,
+  options
+} = {}) =>
+  reduxHelpers.setApiQuery(
+    {
+      ...useProductQuery(),
+      ...useProductQueryFactory(queryType, options)
+    },
+    schemaCheck
+  );
+
+/**
+ * Return an inventory query for hosts.
+ *
+ * @param {object} options
+ * @param {string} options.queryType
+ * @param {object} options.schemaCheck
+ * @param {object} options.options
+ * @returns {object}
+ */
+const useProductInventoryHostsQuery = ({
+  queryType = 'inventoryHostsQuery',
+  schemaCheck = rhsmApiTypes.RHSM_API_QUERY_SET_INVENTORY_TYPES,
+  options
+} = {}) =>
+  reduxHelpers.setApiQuery(
+    {
+      ...useProductQuery(),
+      ...useProductQueryFactory(queryType, options)
+    },
+    schemaCheck
+  );
+
+/**
+ * Return an inventory query for subscriptions.
+ *
+ * @param {object} options
+ * @param {string} options.queryType
+ * @param {object} options.schemaCheck
+ * @param {object} options.options
+ * @returns {object}
+ */
+const useProductInventorySubscriptionsQuery = ({
+  queryType = 'inventorySubscriptionsQuery',
+  schemaCheck = rhsmApiTypes.RHSM_API_QUERY_SET_INVENTORY_TYPES,
+  options
+} = {}) =>
+  reduxHelpers.setApiQuery(
+    {
+      ...useProductQuery(),
+      ...useProductQueryFactory(queryType, options)
+    },
+    schemaCheck
+  );
+
+/**
+ * Get a filtered product configuration context.
+ *
+ * @param {object} options
+ * @param {Function} options.useProductQuery
+ * @param {Function} options.useProductViewContext
+ * @returns {object}
+ */
+const useProductContext = ({
+  useProductQuery: useAliasProductQuery = useProductQuery,
+  useProductViewContext: useAliasProductViewContext = useProductViewContext
+} = {}) => {
+  const { [RHSM_API_QUERY_TYPES.UOM]: uomFilter } = useAliasProductQuery();
+  const {
+    initialGraphFilters = [],
+    initialInventoryFilters = [],
+    initialSubscriptionsInventoryFilters = [],
+    productContextFilterUom,
+    ...config
+  } = useAliasProductViewContext();
+
+  const applyUomFilter = useCallback(() => {
```

Comment:
review: useCallback vs useEffect with a state value may be more in order... we'll keep an eye on it